lib/core/datatable/components/datatable/datatable.component.html

```diff
@@ -60,7 +60,7 @@
                 (select)="onEnterKeyPressed(row, $event)"
                 (keyup)="onRowKeyUp(row, $event)"
                 (keydown)="onRowEnterKeyDown(row, $event)"
-                [adf-upload]="allowDropFiles && rowAllowsDrop(row)"
+                [adf-upload]="((isFolder(row) && allowDropFiles) || (!isFolder(row) && allowDropFileVersion)) && rowAllowsDrop(row)"
```

Comment:
the expression is too long, can you move it to a function and use a single function call in the html? like "canUpload()"